Dockerfile

```diff
@@ -7,46 +7,38 @@ COPY dnf-docker-test/features /behave/
 COPY dnf-docker-test/features /tests/
 COPY rpms /rpms/
 
-RUN echo -e '\necho "deltarpm=0" >> /etc/dnf/dnf.conf' \
+RUN set -x \
     && echo "deltarpm=0" >> /etc/dnf/dnf.conf \
     # behave: core
     # httpd: old-style repos
     # six: py2/py3
     # enum34: rpmdb State
     # whichcraft: shutil.which() for py2
     # jinja2: rpmspec template
-    && echo -e '\ndnf -y install httpd vsftpd python2-behave python2-six python-enum34 python2-whichcraft python-jinja2 python2-pexpect' \
     && dnf -y install httpd vsftpd python2-behave python2-six python-enum34 python2-whichcraft python-jinja2 python2-pexpect \
     && if [ $type = "local" ] ; then \
         # Allows to run test with rpms from only single component in rpms/
-        echo -e '\ndnf -y install dnf-plugins-core python3-dnf-plugins-core python2-dnf-plugins-core rpm-build createrepo_c' \
         && dnf -y install dnf-plugins-core python3-dnf-plugins-core python2-dnf-plugins-core rpm-build createrepo_c  \
-        && echo -e '\ndnf -y copr enable rpmsoftwaremanagement/dnf-nightly' \
         && dnf -y copr enable rpmsoftwaremanagement/dnf-nightly ; \
     fi \
     # prevent installation of dnf-plugins-extras (versionlock, local, torproxy)
-    && echo -e '\nrm /rpms/*extras-versionlock*.rpm /rpms/*extras-local*.rpm /rpms/*extras-torproxy*.rpm' \
-    && rm /rpms/*extras-versionlock*.rpm /rpms/*extras-local*.rpm /rpms/*extras-torproxy*.rpm || echo "RPMs files do not exist" >&2 \
-    && echo -e '\ndnf -y --best upgrade dnf' \
+    && rm -vf /rpms/*extras-versionlock*.rpm /rpms/*extras-local*.rpm /rpms/*extras-torproxy*.rpm \
+    # update dnf
     && dnf -y --best upgrade dnf \
-    && echo -e '\nif ls /rpms/*.rpm 1> /dev/null 2>&1; then dnf -y install /rpms/*.rpm; else echo "RPMs files do not exist" >&2; fi' \
-    && if ls /rpms/*.rpm 1>/dev/null 2>&1; then dnf -y install /rpms/*.rpm else echo "RPMs files do not exist" >&2; fi \
     && if [ $type = "local" ] ; then \
-        # next line is a temporary workaround for RhBug:1398272
-        rpm -q dnf \
-        && echo -e '\ndnf -y mark install $(dnf repoquery --unneeded -q)' \
+        # install all rpms if present
+        if ls /rpms/*.rpm 1>/dev/null 2>&1; then dnf -y install /rpms/*.rpm; fi \
```

Comment:
Still not understand why jjb cannot use 
```if ls /rpms/*.rpm 1>/dev/null 2>&1; then dnf -y install /rpms/*.rpm else echo "RPMs files do not exist" >&2; fi \```
and have to use
``` dnf -y install /rpms/*.rpm ```
The meaning is same, but first one report if there is no rpms. With jjb there is another insurance that there are rpms.